### ISSUE TITLE
intersect_dense_pruned.cu Check failed

### ISSUE THREAD

**Issue by lifeiteng** [k2-fsa/k2#1311]
error trace
```
0: [F] /var/www/k2/csrc/intersect_dense_pruned.cu:1304:lambda [](int)->void::operator()(int)->void block:[65,0,0], thread: [106,0,0] Check failed: backward_loglike < -src_state_forward_loglike + 2.0 (10289668.000000 vs. 10289668.000000) 
0: [F] /var/www/k2/csrc/array.h:385:T k2::Array1<T>::operator[](int32_t) const [with T = int; int32_t = int] Check failed: ret == cudaSuccess (710 vs. 0)  Error: device-side assert triggered. 
0: [F] /var/www/k2/csrc/eval.h:147:void k2::EvalDevice(cudaStream_t, int32_t, LambdaT&) [with LambdaT = __nv_dl_wrapper_t<__nv_dl_tag<void (k2::MultiGraphDenseIntersectPruned::*)(int, intersect_pruned_internal::FrameInfo*, intersect_pruned_internal::FrameInfo*, k2::Array1<char>*, k2::Array1<char>*), &k2::MultiGraphDenseIntersectPruned::PropagateBackward, 2>, intersect_pruned_internal::StateInfo*, int*, int*, const float*, char*, float>; cudaStream_t = CUstream_st*; int32_t = int] Check failed: e == cudaSuccess (710 vs. 0)  Error: device-side assert triggered. 
0: 
0: 
0: [ Stack-Trace: ]
0: /opt/conda/lib/python3.11/site-packages/k2/lib64/libk2_log.so(k2::internal::GetStackTrace()+0x34) [0x7787474419b4]
0: /opt/conda/lib/python3.11/site-packages/k2/lib64/libk2context.so(k2::internal::Logger::~Logger()+0x2a) [0x778747c05d8a]
0: /opt/conda/lib/python3.11/site-packages/k2/lib64/libk2context.so(void k2::EvalDevice<__nv_dl_wrapper_t<__nv_dl_tag<void (k2::MultiGraphDenseIntersectPruned::*)(int, intersect_pruned_internal::FrameInfo*, intersect_pruned_internal::FrameInfo*, k2::Array1<char>*, k2::Array1<char>*), &k2::MultiGraphDenseIntersectPruned::PropagateBackward, 2u>, intersect_pruned_internal::StateInfo*, int*, int*, float const*, char*, float> >(CUstream_st*, int, __nv_dl_wrapper_t<__nv_dl_tag<void (k2::MultiGraphDenseIntersectPruned::*)(int, intersect_pruned_internal::FrameInfo*, intersect_pruned_internal::FrameInfo*, k2::Array1<char>*, k2::Array1<char>*), &k2::MultiGraphDenseIntersectPruned::PropagateBackward, 2u>, intersect_pruned_internal::StateInfo*, int*, int*, float const*, char*, float>&)+0x214) [0x778747dbb264]
0: /opt/conda/lib/python3.11/site-packages/k2/lib64/libk2context.so(k2::MultiGraphDenseIntersectPruned::PropagateBackward(int, intersect_pruned_internal::FrameInfo*, intersect_pruned_internal::FrameInfo*, k2::Array1<char>*, k2::Array1<char>*)+0xa4a) [0x778747dc6cba]
0: /opt/conda/lib/python3.11/site-packages/k2/lib64/libk2context.so(k2::MultiGraphDenseIntersectPruned::PruneTimeRange(int, int)+0x6d1) [0x778747dd12a1]
0: /opt/conda/lib/python3.11/site-packages/k2/lib64/libk2context.so(std::_Function_handler<void (), k2::MultiGraphDenseIntersectPruned::Intersect(k2::DenseFsaVec*)::{lambda()#1}>::_M_invoke(std::_Any_data const&)+0x1e7) [0x778747dd40a7]
0: /opt/conda/lib/python3.11/site-packages/k2/lib64/libk2context.so(k2::ThreadPool::ProcessTasks()+0x163) [0x778747f73313]
0: /opt/conda/bin/../lib/libstdc++.so.6(+0xd3b75) [0x7788554f0b75]
0: /lib/x86_64-linux-gnu/libc.so.6(+0x94ac3) [0x77885641dac3]
0: /lib/x86_64-linux-gnu/libc.so.6(+0x126850) [0x7788564af850]
0: 
```

code at 0: [F] /var/www/k2/csrc/intersect_dense_pruned.cu:1304
<img width="1346" alt="截屏2024-11-27 13 32 46" src="https://github.com/user-attachments/assets/68c5fbd4-8a4f-44c4-895d-79300f400e19">

should it be `K2_CHECK_LE(backward_loglike, -src_state_forward_loglike + 2.0)` ?

**Comment by danpovey** [comment]
Well it certainly wouldn't hurt to change that LT to LE, since the 2.0 is just like an epsilon.  This would likely only make a difference in the case of infinities.  Is it possible to change that, recompile and see if it makes a difference?
There is likely an infinity there and I haven't touched the code recently to know whehter that's supposed to be possible at that point in the code.

**Comment by lifeiteng** [comment]
It is a special audio data, the quality of the data is very poor, so the model logits is also strange.

I'll try compile k2 from source code.

**Comment by danpovey** [comment]
I don't see why that would make the logits strange.  Are these logits after a softmax?

**Comment by lifeiteng** [comment]
The audio data(It sounds like a broken voice in a tornado):
<img width="1663" alt="截屏2024-11-27 16 20 30" src="https://github.com/user-attachments/assets/6a35f420-3424-454d-b06d-509b184eb2d5">

`logits = logits.log_softmax(-1)`
